CesiumGeospatial/include/CesiumGeospatial/SimplePlanarEllipsoidCurve.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "Library.h"
+
+#include <CesiumGeospatial/Ellipsoid.h>
+
+#include <glm/vec3.hpp>
+
+#include <optional>
+
+namespace CesiumGeospatial {
+
+/**
+ * @brief Produces points on an ellipse that lies on a plane that intersects the
```

Comment:
Could add a note that the curve generated is specifically on a WGS 84 ellipse. (in case people think it's more generalized that in actually is)

Reply:
Well, the curve it produces isn't on a WGS84 ellipsoid, since it's floating above the surface. It's a section of an ellipse formed by points on a WGS84 ellipsoid. But I'll clarify that this curve is specific to a WGS84 ellipsoid. Though, actually, it may be better to let the end user specify an ellipsoid so that this supports other ellipsoids right out of the box...

Reply:
Good point. WGS84 ellipsoid inspired?

I did see mentions of WGS84 in the creation functions, so this is probably not a big deal.

Reply:
I've updated SimplePlanarEllipsoidCurve to take an Ellipsoid as input. It's a simple change but it means that it'll support, for example, generating curves between different points on the moon or on mars, if someone wants to do that.

Reply:
Unfortunately we can't currently support passing the ellipsoid in without some changes to how ellipsoids are represented on the Unity side, which is definitely beyond the scope here. So it'll have to be hardcoded to a WGS84 ellipsoid in Unity for now. I'll make a note of that.